relays/messages-relay/src/message_race_delivery.rs

```diff
@@ -369,11 +374,44 @@ impl<P: MessageLane> RaceStrategy<SourceHeaderIdOf<P>, TargetHeaderIdOf<P>, P::M
 				let to_requeue = range
 					.into_iter()
 					.skip_while(|(_, weight)| {
+						// Since we (hopefully) have some reserves in `max_messages_weight_in_single_batch`
+						// and `max_messages_size_in_single_batch`, we may still try to submit transaction
+						// with single message if message overflows these limits. The worst case would be if
```

Comment:
Just to clarify, the worst thing that might happen is:
1. The transaction is rejected from the pool (i.e. not even included in the block)
2. So the relayer will not really be charged fee (i.e. we don't open a griefing vector here)
3. Such thing should never really happen in practice unless runtimes are misconfigured, right? If it happens the lane is essentially clogged, unless relayers are configured incorrectly.

Reply:
Normally yes - afair, before accepting tx to the pool, all its signed extensions are checked by all supported runtimes. And weight and size checks are done by `CheckWeight` extension. So out-of-bounds tx shall not even reach the pool => fee would not be charged.

Such thing should not happen in practice. But in addition to runtime/relayer misconfiguration, there's another case when it might happen - if storage proof is too large. If storage trie is large enough to generate proofs that can occupy 2/3 of maximal tx size, then relay may select e.g. 2 messages for the single delivery tx. Let's say the proof would take the whole tx size (1/3 for messages and (2/3 + 1 byte) for trie nodes). In this case the lane is not actually clogged (because we may deliver messages 1-by-1), but the relayer will always select 2-messages for transaction => messages will never be delivered. This may be fixed in relayer code, but if the proof is that large, then it is actually easier to cheat source runtime, actually clogging the lane.